pkg/services/navtree/navtreeimpl/navtree.go

```diff
@@ -345,22 +345,17 @@ func (s *ServiceImpl) buildStarredItemsNavLinks(c *contextmodel.ReqContext) ([]*
 		return nil, err
 	}
 
-	starredDashboards := []*dashboards.Dashboard{}
-	starredDashboardsCounter := 0
-	for dashboardId := range starredDashboardResult.UserStars {
-		// Set a loose limit to the first 50 starred dashboards found
-		if starredDashboardsCounter > 50 {
-			break
-		}
-		starredDashboardsCounter++
-		query := &dashboards.GetDashboardQuery{
-			ID:    dashboardId,
-			OrgID: c.OrgID,
-		}
-		queryResult, err := s.dashboardService.GetDashboard(c.Req.Context(), query)
-		if err == nil {
-			starredDashboards = append(starredDashboards, queryResult)
-		}
+	var ids []int64
+	for id := range starredDashboardResult.UserStars {
+		ids = append(ids, id)
+	}
+	starredDashboards, err := s.dashboardService.GetDashboards(c.Req.Context(), &dashboards.GetDashboardsQuery{DashboardIDs: ids, OrgID: c.OrgID})
+	if err != nil {
+		return nil, err
+	}
+	// Set a loose limit to the first 50 starred dashboards found
+	if len(starredDashboards) > 50 {
+		starredDashboards = starredDashboards[:50]
```

Comment:
We need to check if we have any ids and only query for dashboards in that case. Now if I have no starred dashboards I get an error and cannot render the page

Reply:
should we add the ids protection (return empty slice and no error when len(ids) == 0) inside of `dashboardService.GetDashboards` to avoid it happening elsewhere

Reply:
@IevaVasiljeva sure thing